common/src/main/java/gov/cms/ab2d/common/util/FilterOutByDate.java

```diff
@@ -266,7 +266,8 @@ static boolean afterDate(Date dateVal, ExplanationOfBenefit ben) throws ParseExc
         if (ben == null || ben.getBillablePeriod() == null || dateVal == null) {
             return false;
         }
-        Date attToUse = fullDateFormat.parse(shortDateFormat.format(dateVal) + " 00:00:00");
+
+        Date attToUse = fullDateFormat.parse(shortDateFormat.format(dateVal) + " 00:00:00:000");
         Period p = ben.getBillablePeriod();
```

Comment:
Why not just add a method on DateRange that takes a Period and returns a boolean if there is a match?